x-pack/test/functional/services/ml/data_visualizer_table.ts

```diff
@@ -61,21 +72,53 @@ export function MachineLearningDataVisualizerTableProvider({ getService }: FtrPr
       await testSubjects.existOrFail(this.rowSelector(fieldName));
     }
 
-    public async expandRowDetails(fieldName: string, fieldType: string) {
+    public detailsSelector(fieldName: string, subSelector?: string) {
+      const row = `~mlDataVisualizerTable > ~mlDataVisualizerFieldExpandedRow-${fieldName}`;
+      return !subSelector ? row : `${row} > ${subSelector}`;
+    }
+
+    public async ensureDetailsOpen(fieldName: string) {
+      await retry.tryForTime(10000, async () => {
+        if (!(await testSubjects.exists(this.detailsSelector(fieldName)))) {
+          await testSubjects.click(this.rowSelector(fieldName, 'mlDataVisualizerDetailsToggle'));
+          await testSubjects.existOrFail(
+            this.rowSelector(fieldName, `mlDataVisualizerDetailsToggle-${fieldName}-arrowUp`),
+            {
+              timeout: 1000,
+            }
+          );
+          await testSubjects.existOrFail(this.detailsSelector(fieldName), { timeout: 1000 });
+        }
+      });
+    }
+
+    public async ensureDetailsClosed(fieldName: string) {
+      await retry.tryForTime(10000, async () => {
+        if (await testSubjects.exists(this.detailsSelector(fieldName))) {
+          await testSubjects.click(this.rowSelector(fieldName, 'mlDataVisualizerDetailsToggle'));
+          await testSubjects.existOrFail(
+            this.rowSelector(fieldName, `mlDataVisualizerDetailsToggle-${fieldName}-arrowDown`),
+            {
+              timeout: 1000,
+            }
+          );
+          await testSubjects.missingOrFail(this.detailsSelector(fieldName), { timeout: 1000 });
+        }
+      });
+    }
+
+    public async openDetails(fieldName: string) {
```

Comment:
I think this method is not needed any more. To open the details we can call `ensureDetailsOpen` which will do all the needed checks and clicks.
Also, I think we should call `ensureDetailsClosed` in the test file, when we're the row expansion content validation is done.